src/provider/provider_os_memory.c

```diff
@@ -308,39 +291,42 @@ static void os_finalize(void *provider) {
 static umf_result_t os_get_min_page_size(void *provider, void *ptr,
                                          size_t *page_size);
 
+#ifndef NDEBUG
 static void print_numa_nodes(os_memory_provider_t *os_provider, void *addr,
                              size_t size) {
 
+    // This function do not use logger as it is expected to print a message
+    // to stdout/stderr
```

Comment:
do not use logger? I see LOG_DEBUG few next lines below

Reply:
Done.